### ISSUE TITLE
(Fix Guide) Paperspace Python3.9 is missing

### ISSUE THREAD

**Issue by fartist123** [TheLastBen/fast-stable-diffusion#2834]
This is a simple guide to repair the following paperspace error:

![error](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/4da48f4d-a557-49fc-a07c-19e70d76650c)

I still dont understand why the issue only started recently, but here is how to solve it:

First, open a new terminal on the paperspace console:

![term1](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/084f4876-7eef-44ca-8126-7b21c805f9ad)
![term2](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/5db11de0-17b1-4d58-8a4a-2837122a1833)
![term3](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/ff194eb6-ca37-4fb8-9fcc-d3d7a9e86024)

Then paste the following into the Termal prompt (all the text in one prompt):

sudo apt update
sudo apt install software-properties-common
sudo add-apt-repository -y ppa:deadsnakes/ppa
sudo apt-get -y install python3.9 python3.10 python3.11 python3.12

Now run the following Code-cells in this order:

Run the Dependencies Cell
Run Install/Update Automatic repo Cell
Run Model Cell
Run LoRA Cell
Run ControlNet Cell

Then paste the following into the Termal prompt (all the text in one prompt):

python -m pip install pytorch-lightning
pip install torchvision==0.15.2 torchaudio==2.0.2 tensorrt
!pip install tensorflow[and-cuda] git+https://github.com/openai/CLIP.git
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib/python3.11/dist-packages/tensorrt_libs/
pip3 install accelerate==0.20.3
pip3 install -r /notebooks/sd/stable-diffusion-webui/requirements_versions.txt 
pip install onnx numexpr polygraphy onnxruntime onnx_graphsurgeon optimum
pip install --force-reinstall --no-deps --pre xformers==0.0.20
set COMMANDLINE_ARGS= --allow-code --xformers --skip-torch-cuda-test --no-half-vae --api --listen --enable-insecure-extension-access --share

Then add the following commands to your "Start Stable-Diffusion" code-cell:

--allow-code --xformers --skip-torch-cuda-test --no-half-vae --api --listen --enable-insecure-extension-access --share

![add](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/0e02748d-3f67-4d9e-aa48-c6ca29783c78)
![add2](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/5e8b85c1-9316-4d35-ab82-f65bffc12e96)
![add3](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/99ff1d42-ab7f-428e-bfe3-18149266ae7e)

Then Run the "Start Stable-Diffusion" code-cell and it should boot.

Your welcome.

Checkout my youtube channel if you feel interested:
https://www.youtube.com/channel/UCHTWpAApt4alEyO-jLVlXMg

**Comment by kocten7** [comment]
Unfortunately, it didn't work out for me.

![PS](https://github.com/TheLastBen/fast-stable-diffusion/assets/79367127/1b86f567-6c56-4eff-942d-4088bb154303)

It would be great if the author fast-stable-diffusion or Paperspace would fix this. I'm not good at coding.

**Comment by fartist123** [comment]
pip3 install -r /notebooks/sd/stable-diffusion-webui/requirements_versions.txt

that command includes the installation of gradio, try pasting all the lines into the prompt rather then just one at a time.

And I'm not good at coding either, I literally learnt linux 3 days ago if you can believe it.

**Comment by kocten7** [comment]
Thanks for the support!

```
Run the Dependencies Cell
Run Install/Update Automatic repo Cell
Run Model Cell
Run LoRA Cell
Run ControlNet Cell
```
I'm having trouble at this step:
![term](https://github.com/TheLastBen/fast-stable-diffusion/assets/79367127/66a5aa9b-9787-4db1-aae2-1cebde7d5d8b)

**Comment by kocten7** [comment]
I got it! It's incredible!
I pasted that line into the terminal and everything seemed to work:
`pip3 install -r /notebooks/sd/stable-diffusion-webui/requirements_versions.txt`

Thank you so much for your help!

**Comment by fartist123** [comment]
Run the Dependencies Cell

![run1](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/57b33111-fd3e-4bee-9d03-29501a2e89d8)

Run Install/Update Automatic repo Cell

![run2](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/5fb87cf3-5c48-4b24-aa8b-b0d5a429bdb1)

Run Model Cell

![run3](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/eb860eab-72c2-4a33-8cc3-87a4d5f5149e)

Run LoRA Cell

![run4](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/a6b1a37f-c9c5-47e2-89c0-a0731e9006f4)

Run ControlNet Cell

![run5](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/78e02066-6dd9-47cf-96be-e8f58fbf369d)

Check out my youtube channel, you owe me.
https://www.youtube.com/channel/UCHTWpAApt4alEyO-jLVlXMg

**Comment by kocten7** [comment]
I subscribed to your YT channel and will be watching some videos soon. 🤝

**Comment by TheLastBen** [comment]
this issue happens only if you choose a different starting template, most users don't face this issue because they start with the paperspace pytorch 1.12 template

**Comment by fartist123** [comment]
I've literally tried all the templates, including starting from scratch, and they all load with Python3.9 missing. 

My issues started after a "update" button appeared next to this:

![versiontype](https://github.com/TheLastBen/fast-stable-diffusion/assets/115401859/682d169b-8ce1-4b8e-a87b-71ecd3d78677)

Ever since I pressed the update, no matter what template I choose, this problem is persists. 

The reason I made this guide in the first place is I'm fully aware that if I'm having this problem, then many other people are having the exact problem as well, statistically. As they say, "there is nothing new under the sun". 

I tried all the obvious fixes, nothing worked. Please assume I tried pytorch 1.12 template. The lack of support I've had with this problem drove me to learn basic Linux, and boy, I dont envy coders... not that I count anything in my guide as code.

The issue lies within paperspaces default gradient image, this I swear.

**Comment by TheLastBen** [comment]
OK, I'll check it out

**Comment by TheLastBen** [comment]
The problem has been fixed, everything works with the new python 3.11, no need to install anything, just make sure you use the latest notebooks, by copying them from the folder "latest_notebooks"

**Comment by fartist123** [comment]
Very much appreciated! Your doing the lords work!

**Comment by kocten7** [comment]
I can't believe it's working! Thank you very much for fixing this issue, @TheLastBen
